ERCS/erc-0626.md

```diff
@@ -0,0 +1,119 @@
+The interface includes methods to retrieve the most recent JSON state as well as the version history of each key in the JSON object. 
+This approach supports REST developers familiar with JSON-based data interactions, thus improving accessibility for developers new to Web3 and Ethereum.
+
+## Motivation
+
+With an increasing number of developers from RESTful backgrounds joining the Ethereum ecosystem, there is a need for a contract interface that allows developers to easily interact with structured JSON data. 
+This EIP aims to create a universal standard that provides JSON data management and version control functionality in a straightforward and REST-like way, making Ethereum more accessible and developer-friendly.
+
+## Specification
+
+The contract interface includes the following methods:
+
+### Read Methods
+
+1. **`json()`**
+
+   - **Inputs:** None.
+   - **Output:** `string` (a JSON object as a string)
+   - **Description:** Returns the JSON object in its latest state. The output uses the most recent value for each key in the JSON object, mimicking a REST JSON call.
+   - **Behavior:** Each key in the JSON object reflects the latest "Last In, First Out" (LIFO) state of its value.
+
+2. **`version(string key)`**
+
+   - **Inputs:** 
+     - `key` (`string`): The JSON key whose version history is requested.
+   - **Output:** `string` (a JSON array as a string)
+   - **Description:** Returns an array of all versions of the specified key's value in JSON format. 
+     The array is ordered chronologically, with the earliest version at index 0 and the most recent version at the highest index.
+
+### Write Method
+
+1. **`write(string[] keys, string[] values, bool replace)`**
+
+   - **Inputs:**
+     - `keys` (`string[]`): Array of JSON key strings, where each key represents a path within the JSON object.
+     - `values` (`string[]`): Array of JSON-compatible strings corresponding to each key. Can represent primitive types, arrays, or objects.
+     - `replace` (`bool`): Determines whether to replace an existing key's latest value.
+       - If `replace` is `true` and the key exists, the existing value is updated in the `json` output.
+       - However, the previous value persists in the `version` output, with the new value appended at the end of the version history array.
+       - If `replace` is `false` and the key already exists, the transaction reverts.
+   - **Output:** None.
+   - **Description:** Writes new values to the JSON object, either adding a new key or updating an existing one, based on the `replace` parameter.
+
+#### Additional Requirements
+
+- **Universal ABI:** To support widespread adoption and ease of use, the ABI should be standardized for REST-like abstraction within popular Web3 libraries. 
+  By making this contract accessible through Web3 abstractions, developers with REST experience can seamlessly interact with the contract.
+  
+## Rationale
+
```

Comment:
You'll definitely want to explain why you're returning JSON strings instead of providing an key-value interface.